File: Day8_Stock Buy and Sell – Max one Transaction Allowed/stock_buy_sell.py
# https://www.geeksforgeeks.org/batch/gfg-160-problems/track/arrays-gfg-160/problem/buy-stock-2
def maximumProfit(prices):
    # code here
    buy = prices[0]
    profit = 0
    for i in range(1,len(prices)):
        if prices[i] < buy:
            buy = prices[i]
        elif prices[i] - buy > profit:
            profit = prices[i] - buy
    return profit


user_input = input("Enter space-separated integers: ")
arr = list(map(int, user_input.split()))
print(maximumProfit(arr=arr))

File: Day8_Stock Buy and Sell – Max one Transaction Allowed/stock_buy_sell.java
import java.util.ArrayList;
import java.util.Scanner;

public class stock_buy_sell {
    public static int maximumProfit(ArrayList<Integer> arr) {
        int buy = arr.get(0);
        int profit = 0;
        for (int i = 1; i < arr.size(); i++) {
            if(arr.get(i)<buy)
                buy = arr.get(i);
            else if(arr.get(i)-buy > profit)
                profit = arr.get(i)-buy;
        }
        return profit;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the size and element of array: ");
        int n = sc.nextInt();
        ArrayList<Integer> arr = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            arr.add(sc.nextInt());
        }
        int result = maximumProfit(arr);
        System.out.print("Maximum profit: "+ result);
        System.out.println();
        sc.close();
    }
}
